From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 27 Mar 2022 12:30:33 +0200
Subject: [PATCH] {MAYBE UPS} drivers/net/ethernet/fungible/funcore: fix
 division for 32-bit architectures

Building on ARM (32 bits) with gcc fails:

    ERROR: modpost: "__aeabi_uldivmod" [drivers/net/ethernet/fungible/funcore/funcore.ko] undefined!

Introduce a possibly-buggy cast to work around this issue.
---
 drivers/net/ethernet/fungible/funcore/fun_dev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/fungible/funcore/fun_dev.c b/drivers/net/ethernet/fungible/funcore/fun_dev.c
index 5d7aef73df61..fb3ef9a04ddc 100644
--- a/drivers/net/ethernet/fungible/funcore/fun_dev.c
+++ b/drivers/net/ethernet/fungible/funcore/fun_dev.c
@@ -586,7 +586,7 @@ static int fun_get_dev_limits(struct fun_dev *fdev)
 	/* Calculate the max QID based on SQ/CQ/doorbell counts.
 	 * SQ/CQ doorbells alternate.
 	 */
-	num_dbs = (pci_resource_len(pdev, 0) - NVME_REG_DBS) /
+	num_dbs = ((unsigned int)pci_resource_len(pdev, 0) - NVME_REG_DBS) /
 		  (fdev->db_stride * 4);
 	fdev->max_qid = min3(cq_count, sq_count, num_dbs / 2) - 1;
 	fdev->kern_end_qid = fdev->max_qid + 1;
-- 
